### DISCUSSION TITLE
feat: loadbalance multiple ollama servers in kubernetes

### DISCUSSION THREAD

**Discussion by keisni** [open-webui/open-webui#2383] [Q&A]
Maybe this should be regarded as a question. 

I deployed open-webui & ollama with the helm chart in the source code, and I want to make sure If I run more than one replicas of the ollama server, they can be well loadbalanced and don`t cause any inconsistence. 

Clearly, these ollama servers will be loadbalanced through cluster service. Does this mean that if I upload a model, it will be upload to any of them? And then if the chat request routed to a server without the loaded model, I will see a error reply? I known that there is a "Update all models" button, but I`m not sure this still works through the single clusterIP.

**Comment by keisni** [comment]
the helm chart in kubernetes directory create a Statefulset and CusterIP service for ollma. Through a single url like http://ollama-service.open-webui.svc.cluster.local:11434, open-webui lost tracking of which instance have which model. So I think this is a mistake. Use a headless service with statefulset is common pratice in k8s, and open-webui can fetch all ollama instance ips throuch dns resolve. Did I miss something?

**Comment by Expro** [comment]
Works fine as-is. You can share volume across multiple instances of ollama, so that modle uploaded to one is visible to all of the instances.

**Reply by keisni** [reply]
> Works fine as-is. You can share volume across multiple instances of ollama, so that modle uploaded to one is visible to all of the instances.

I don`t know the conditions below could really happen.

Suppose there are two ollama server instances running.
Now we have two users, Alice and Bob.
Alice selected model llama2:7b and Bob selected model qwen:14b.
When they are talking to LLMs, there requests are random distriubted cross the two ollama instances.
Each ollama server can access only one GPU and has one runner.
So the ollama servers are swapping the two models all the time.

**Comment by AdaptiveStep** [comment]
Anyone got any idea on how to limit this properly for multiple users?
